src/qtt/measurements/scans.py

```diff
@@ -1948,6 +1946,21 @@ def acquire_segments(station, parameters, average=True, mV_range=2000,
 
     return alldata
 
+def _is_m4i(instrument):
+    try:
+        is_m4i = isinstance(instrument, qcodes.instrument_drivers.Spectrum.M4i.M4i)
+        if is_m4i:
+               warnings.warn('please use qcodes_contrib_drivers.drivers.Spectrum.M4i instead of qcodes.instrument_drivers.Spectrum.M4i.M4i)')
+               return is_m4i
+    except:
```

Comment:
you should not use a bare except: when not needed
Depending on an earlier comment this should be changed to catch only TypeError